GeneralisedFilters/src/algorithms/kalman.jl

```diff
@@ -129,6 +131,58 @@ function update(
     return BatchGaussianDistribution(μ_filt, Σ_filt), dropdims(log_likes; dims=1)
 end
 
+## SQUARE-ROOT KALMAN FILTER ###############################################################
+
+"""
+     SRKF()
+
+     A square-root Kalman filter.
+
+     Implemented by wrapping KalmanFilters.jl.
+ """
+struct SRKF <: AbstractFilter end
+
+function initialise(
+    rng::AbstractRNG, model::LinearGaussianStateSpaceModel, filter::SRKF; kwargs...
+)
+    μ0, Σ0 = calc_initial(model.dyn; kwargs...)
+    return Gaussian(μ0, Σ0)
+end
+
+function predict(
+    rng::AbstractRNG,
+    model::LinearGaussianStateSpaceModel,
+    ::SRKF,
+    iter::Integer,
+    state::Gaussian,
+    observation=nothing;
+    kwargs...,
+)
+    μ, Σ = GaussianDistributions.pair(state)
+    A, b, Q = calc_params(model.dyn, iter; kwargs...)
+    !all(b .== 0) && error("SKRF doesn't current support non-zero b")
+
+    tu = KalmanFilters.time_update(μ, cholesky(Σ), A, cholesky(Q))
+    return Gaussian(tu.state, PDMat(tu.covariance))
+end
+
+function update(
+    model::LinearGaussianStateSpaceModel,
+    algo::SRKF,
+    iter::Integer,
+    state::Gaussian,
+    observation::AbstractVector;
+    kwargs...,
+)
+    μ, Σ = GaussianDistributions.pair(state)
+    H, c, R = calc_params(model.obs, iter; kwargs...)
+    !all(c .== 0) && error("SKRF doesn't current support non-zero c")
```

Comment:
```suggestion
    c != zero(μ) && error("SKRF doesn't current support non-zero c")
```
I doubt this gets us any speed up, but should guarantee type stability in 99% of cases.